### PR TITLE
tests, infra-test, Adapt infra_test to support dual stack

### PR DESCRIPTION
**Pull request by oshoval**
Adapt tests in infra_test.go to check dual stack if exists using `DescribeTable`.

```release-note
None
```

### CODE REVIEW

File: tests/libnet/skips.go [hunk 1]
```diff
@@ -14,3 +14,8 @@ func SkipWhenNotDualStackCluster(virtClient kubecli.KubevirtClient) {
 		Skip("This test requires a dual stack network config.")
 	}
 }
+
+// func IsRunningOnKindInfraIPv6() bool {
```
Comment (oshoval) [inline tests/libnet/skips.go:18]: to remove

**Comment by oshoval** [comment]
/hold
since this PR depends on 2 other PRs (cherry-picking them until they are merged), 
see notes in the PR description please which are unique for this PR.

**Comment by oshoval** [comment]
Just rebased, didnt take new 4306 yet. because the changes there still dont affect the unique commit of this PR.

**Comment by oshoval** [comment]
Recherry-picked 4306, tested the new logic works both when IPv6 is skipped or when its not (to see it detect the error)

**Comment by oshoval** [comment]
Recherry-picked 4306

**Comment by AlonaKaplan** [comment]
Please add description to the PR.

**Comment by oshoval** [comment]
> Please add description to the PR.

Done, once 4306 will be lgtm approve, i will cherry-pick again.

File: tests/infra_test.go [hunk 2]
```diff
@@ -984,3 +1037,19 @@ func validatedHTTPResponses(errorsChan chan error, concurrency int) error {
 
 	return sampledError
 }
+
+func getSupportedIP(virtClient kubecli.KubevirtClient, ips []string, family k8sv1.IPFamily) string {
+	ip := libnet.GetIp(ips, family)
+	if ip == "" {
+		if family == k8sv1.IPv4Protocol {
+			if tests.IsRunningOnKindInfraIPv6() {
```
Comment (AlonaKaplan) [inline tests/infra_test.go:1045]: We should remove this method and all it usages. We don't support this cluster any more. Please don't use it.

File: tests/infra_test.go [hunk 3]
```diff
@@ -984,3 +1037,19 @@ func validatedHTTPResponses(errorsChan chan error, concurrency int) error {
...
+				Skip("Skip IPv4 on Single stack IPv6")
+			}
+		} else {
+			libnet.SkipWhenNotDualStackCluster(virtClient)
```
Comment (AlonaKaplan) [inline tests/infra_test.go:1049]: It is enough to check whether the requested family is ipv6.
```
	if family == k8sv1.IPv6Protocol {
						libnet.SkipWhenNotDualStackCluster(virtClient)
					}
```
For better readability of the tests, I prefer to move the skip logic to the beginning of the `It`.
In `getSupportedIP` you should assume there is an ip. If `ip==""` you should fail.

File: tests/infra_test.go [hunk 4]
```diff
@@ -682,6 +693,8 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			}
 
 			errorsChan := make(chan error)
+			By("Scraping the Prometheus endpoint")
+			metricsURL := fmt.Sprintf("https://%s/metrics", net.JoinHostPort(ip, strconv.Itoa(8443)))
```
Comment (AlonaKaplan) [inline tests/infra_test.go:697]: Please extract to a method. I duplicate [this](https://github.com/kubevirt/kubevirt/pull/4307/commits/952eb7f1d9b59c9253de87f5068aedf4a9d24844#diff-c95f11746affb831bc43f837f8be65ad753336c090e3b181531ec00bf7636d7dR452).

**Comment by oshoval** [comment]
Recherry-pick 4306

**Comment by oshoval** [comment]
Answered comments, PTAL

**Comment by oshoval** [comment]
Rebased, just in order to update tto the merged hashes
@RamLavi can you please re-lgtm ?

**Pull request merged**